### PR TITLE
implement jwt-auth

### PR DESCRIPTION
**Pull request by steveklabnik**
WIP, don't merge

### CODE REVIEW

File: app/authenticators/custom.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import Ember from 'ember';
+import Base from 'simple-auth/authenticators/base';
+
+export default Base.extend({
+  tokenEndpoint: 'http://localhost:3001/sessions/create',
```
Comment (ashleygwilliams) [inline app/authenticators/custom.js]: port 8080

File: app/authenticators/custom.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  },
+
+  authenticate: function(options) { 
+    var self = this;
+    return new Ember.RSVP.Promise(function(resolve, reject) { 
+    Ember.$.ajax({ 
+      url: self.tokenEndpoint, 
+      type: 'POST', 
+      data: JSON.stringify({
+          username: options.identification,
```
Comment (ashleygwilliams) [inline app/authenticators/custom.js]: email

File: app/router.js [hunk 3]
```diff
@@ -9,7 +9,8 @@ Router.map(function() {
   this.resource('customers', function() {
     this.route('new');
   });
-  this.route('login', { path: '/login' });
+  this.route('login');
```
Comment (ashleygwilliams) [inline app/router.js:12]: do we not still need the path?

File: app/router.js [hunk 4]
```diff
@@ -9,7 +9,8 @@ Router.map(function() {
...
+  this.route('protected');
```
Comment (ashleygwilliams) [inline app/router.js:13]: not using this...

Comment (ashleygwilliams) [inline app/router.js:13]: protected is all the views except home and login

File: app/templates/components/login-form.hbs [hunk 5]
```diff
@@ -0,0 +1,21 @@
+<div class="row">
```
Comment (ashleygwilliams) [inline app/templates/components/login-form.hbs]: we already have a login template?

Comment (ashleygwilliams) [inline app/templates/components/login-form.hbs]: oh i see. nevermind.

File: bower.json [hunk 6]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "assistant-frontend",
   "dependencies": {
-    "ember": "1.13.2",
+    "ember": "2.0.0-beta.3",
```
Comment (ashleygwilliams) [inline bower.json]: probably a p intense thing to do in this PR... maybe separate?

File: config/environment.js [hunk 7]
```diff
@@ -27,6 +27,13 @@ module.exports = function(environment) {
     }
   };
 
+  ENV['simple-auth'] = {
+    store: 'simple-auth-session-store:local-storage',
+    authorizer: 'authorizer:custom',
+    crossOriginWhitelist: ['http://localhost:3001/'],
```
Comment (ashleygwilliams) [inline config/environment.js]: port 8080

File: app/authenticators/custom.js (hunk 2 above)
Comment (steveklabnik) [inline app/authenticators/custom.js]: i fixed this in the form

File: app/router.js (hunk 3 above)
Comment (steveklabnik) [inline app/router.js:12]: I don't think we ever needed it, and the example doesn't include it

File: app/router.js (hunk 4 above)
Comment (steveklabnik) [inline app/router.js:13]: yes, this is just a bootstrapping thing

File: app/authenticators/custom.js (hunk 2 above)
Comment (ashleygwilliams) [inline app/authenticators/custom.js]: p sure you still need to change this.

Comment (steveklabnik) [inline app/authenticators/custom.js]: ahh, i did not. i thought you were talking about the `identification` part, not the `username` part.

**Comment by ashleygwilliams** [comment]
squash and then r+

**Comment by ashleygwilliams** [comment]
also lint plz

**Comment by steveklabnik** [comment]
:+1:

**Comment by ashleygwilliams** [comment]
:dolphin: :dog: :moneybag:

**Comment by steveklabnik** [comment]
:shipit:

**Pull request merged**